### PR TITLE
Added structure for default values

### PR DESCRIPTION
**Pull request by wimrijnders**
This adds the initial defaults structure. It currently only does the simple fields; the rest require a bit more work.

Sorry, my mind is not clear enough to do the rest right now. Will add it bit by bit.

### CODE REVIEW

File: lib/graph3d/Graph3d.js [hunk 1]
```diff
@@ -147,15 +210,16 @@ function Graph3d(container, data, options) {
   // TODO: customize axis range
 
   // colors
-  this.axisColor = '#4D4D4D';
-  this.gridColor = '#D3D3D3';
   this.dataColor = {
```
Comment (mojoaxel) [inline lib/graph3d/Graph3d.js]: Why not move `dataColor` also to the DEFAULTS object?

File: lib/graph3d/Graph3d.js [hunk 2]
```diff
@@ -85,45 +152,41 @@ function Graph3d(container, data, options) {
...
-  var passValueFn = function(v) { return v; };
-  this.xValueLabel = passValueFn;
-  this.yValueLabel = passValueFn;
-  this.zValueLabel = passValueFn;
-  
-  this.filterLabel = 'time';
-  this.legendLabel = 'value';
   this.showLegend  = undefined; // auto by default (based on graph style)
 
   this.style = Graph3d.STYLE.DOT;
```
Comment (mojoaxel) [inline lib/graph3d/Graph3d.js]: What's with `showLegend` an `style`?

File: lib/graph3d/Graph3d.js (hunk 1 above)
Comment (wimrijnders) [inline lib/graph3d/Graph3d.js]: If you wait a moment, I can get `xCenter` and `yCenter` into the defaults.

**Comment by wimrijnders** [comment]
Here you go; these were easy enough to do.

**Comment by wimrijnders** [comment]
Look, I see the 'Changes requested' message; I hope you understand that the rest is a bit harder to do than simply copying over. I stated in the initial message that this change does the 'simple' fields only, and that the rest requires 'a bit more work'.

Let me put it this way: do you agree until now? If so, there are two options:

1. Accept this PR and let me add the rest of the defaults incrementally in separate PR's.
2. Leave this PR open and wait till the rest has been done.

Both are fine with me. I have a personal preference for incremental PR's, but keeping this one open works for me as well.

File: lib/graph3d/Graph3d.js (hunk 2 above)
Comment (wimrijnders) [inline lib/graph3d/Graph3d.js]: showLegend is non-trivial:

```js
  this.showLegend        =  (this.defaultShowLegend !== undefined) ? this.defaultShowLegend : isLegendGraphStyle;
```

**Comment by wimrijnders** [comment]
I would appreciate a response to previous in due time.

Also, I have to say I am a bit miffed that you slapped a 'Changes requested' on this PR without actually requesting changes. I really don't want to have to guess your thoughts.

Could you please explicitly state what changes you request? Thanks.

**Comment by wimrijnders** [comment]
So. You approved without adding a single word of response.

Alexander, I will state this explicitly because it is bothering me: I consider this non-communication of yours very, very bad form. Please change this.

**Comment by mojoaxel** [comment]
> I consider this non-communication of yours very, very bad form. Please change this.

I'm very sorry! I'll try to communicate more.
Let me just say, it was late yesterday and actually I just agreed with you: *"Accept this PR and let me add the rest of the defaults incrementally in separate PR's"*. That is why I just approved it. I should have just added "comments" and not "request changes". My mistake.
No bad atmosphere intended! :peace_symbol:

**Comment by wimrijnders** [comment]
Thank you very much! I honestly appreciate this response. Time for emoticons:

:+1: :sunny: :smile_cat: :musical_note: :beers:

**Comment by mojoaxel** [comment]
By the way: You are welcomed in our new chat-room (that sounds sooo 90s): https://gitter.im/vis-js

**Comment by wimrijnders** [comment]
Thank you, Unfortunately, the link is not working.

**Comment by mojoaxel** [comment]
> Thank you, Unfortunately, the link is not working.

I fixed it

**Pull request merged**